packages/fxa-settings/src/components/App/index.tsx

```diff
@@ -353,6 +351,12 @@ export const App = ({
     return <LoadingSpinner fullScreen />;
   }
 
+  // If we're on settings route but user is not signed in, redirect immediately
+  if (window.location.pathname?.includes('/settings') && !isSignedIn) {
```

Comment:
LGTM – do we need to confirm there are no paths (now or in the future) that use a path like `/something/settings`?  Or are we guaranteed this component won't ever live on another path like that? 

Are we able to just do an equality like `window.location.pathname === '/settings'` or does that miss too many cases?